File: seo-product/backend/app/schemas/project.py
from pydantic import BaseModel
from datetime import datetime


class ProjectCreate(BaseModel):
    domain: str
    niche: str | None = None


class ProjectResponse(BaseModel):
    id: str
    user_id: str
    domain: str
    niche: str | None
    brand_context: dict
    created_at: datetime
    updated_at: datetime


class CompetitorCreate(BaseModel):
    domain: str


class CompetitorResponse(BaseModel):
    id: str
    project_id: str
    domain: str
    dr_score: int | None
    indexed_pages: int | None
    auto_detected: bool
    created_at: datetime

File: seo-product/frontend/src/types/project.ts
export interface Project {
  id: string;
  user_id: string;
  domain: string;
  niche: string | null;
  brand_context: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface Competitor {
  id: string;
  project_id: string;
  domain: string;
  dr_score: number | null;
  indexed_pages: number | null;
  auto_detected: boolean;
  created_at: string;
}
